### DISCUSSION TITLE
Game Plan for Overtime

### DISCUSSION THREAD

**Discussion by NavigSB** [gp-seattle/rms-app#208] [General]
# Let's Do This!

Alright, here's a basic plan and order for things to do. A lot of the UX is out of order, since it can be done independent of Frontend, but we'll deal with that later :) 

---------------------------------

## Beginning: 

Housecleaning - 
&nbsp;&nbsp;~~Fix Wedgwood spelling error that is breaking the borrowing system~~
&nbsp;&nbsp;~~Cleanup Inventories into one generic Inventory component~~
&nbsp;&nbsp;~~Hook up Annie's modals and toast~~
&nbsp;&nbsp;&nbsp;&nbsp;~~For all current actions (Borrow, Return, Add, Delete) when the server says the call was successful~~
&nbsp;&nbsp;Show Item count in Inventory, like Figma shows
&nbsp;&nbsp;~~Change Inventory not to filter by separate locations, but to sort by certain qualities, like Figma shows~~
&nbsp;&nbsp;Change borrowing to look a bit more like Figma 
&nbsp;&nbsp;&nbsp;&nbsp;(with checkboxes on singles and a numerical selector for multiples, but on subitem inventory pages)
&nbsp;&nbsp;&nbsp;&nbsp;Modal that gives toast and sends back to Dashboard after
&nbsp;&nbsp;Add Furniture category to Add Item
&nbsp;&nbsp;Create a constants slice in Redux to give global access to certain values, such as "locations".

Discussions - 
&nbsp;&nbsp;~~Figure out if backend and UX are simpatico, and what to change to make them so.~~

Testing -
&nbsp;&nbsp;Write tests for if calling backend APIs return proper error messages for invalid cases (For example, will trying to reserve the same item twice during the same time range return the proper error message?)

Backend -
&nbsp;&nbsp;no proper error message being displayed when invalid operation texted
&nbsp;&nbsp;"optional" fields for AddItem etc. not optional (cannot text blank text)

---------------------------------

## Middle:

&nbsp;&nbsp;Item Profile Page

Lists - 
&nbsp;&nbsp;List Profile Page
&nbsp;&nbsp;&nbsp;&nbsp;With new image-taking capabilities
&nbsp;&nbsp;Borrow List Page
&nbsp;&nbsp;Create List Page
&nbsp;&nbsp;Be able to create list from checkout batch 
&nbsp;&nbsp;&nbsp;&nbsp;(so make pure-local lists based on a single checkout by user. Maybe give it its own category on the list page?)
&nbsp;&nbsp;Create List Redux
&nbsp;&nbsp;Hook up list stream with Redux Slice
&nbsp;&nbsp;Implement Dashboard group sorting
&nbsp;&nbsp;Make some way to return whole lists, even when you only borrowed part of the list (list profile page?)
	
Reservations - 
&nbsp;&nbsp;Create Dashboard Reservation Modal
&nbsp;&nbsp;Figure out UX reservations (should it be on items, or lists, or what?)
&nbsp;&nbsp;Create Reservation Redux Slice
&nbsp;&nbsp;Show Reservations on whatever page UX says
&nbsp;&nbsp;Selectable calendar?

Pictures - 
&nbsp;&nbsp;Picture Selection Page, to be hooked up to Item Creation and Item Profile pages
&nbsp;&nbsp;Figure out how to store images on backend
	
Favorites - 
&nbsp;&nbsp;Add Favorites Redux Slice? (More efficient?)

---------------------------------	

## Finishing Touches:

Auth - 
&nbsp;&nbsp;Create Login page
&nbsp;&nbsp;Create loading animation and hook it up to Login page
&nbsp;&nbsp;Make account page

Wrapping All Together - 
&nbsp;&nbsp;Implement favorites list, both on Dashboard and Inventory
&nbsp;&nbsp;Implement the lists inventory, which stays on the main Inventory page
&nbsp;&nbsp;Implement the reservations inventory, which stays on the main Inventory page

**Comment by DRFredrick73** [comment]
Should also make a component for the "locations" variable, so we only have to change it once and all the drop downs change. Thinking from Inventory, Borrow Item, and New Item. The issue I'm running into at the moment is keeping the "All Items" option everywhere except in the New Items page

**Reply by NavigSB** [reply]
Got it on the plan :)
